addons/pylons/functions/fnc_showDialog.sqf

```diff
@@ -0,0 +1,139 @@
+
+params ["_aircraft", ["_isCurator", false]];
+
+if (!GVAR(enabled) || {!(typeOf _aircraft in GVAR(aircraftWithPylons))}) exitWith {};
+
+private _currentUser = _aircraft getVariable [QGVAR(currentUser), objNull];
+if (!isNull _currentUser) exitWith {
+    [format [localize LSTRING(InUse), name _currentUser], false, 5] call EFUNC(common,displayText);
+};
+_aircraft setVariable [QGVAR(currentUser), ace_player, true];
+GVAR(currentAircraftNamespace) setVariable [getPlayerUID ace_player, _aircraft, true];
+[_aircraft, "blockEngine", QUOTE(ADDON), true] call EFUNC(common,statusEffect_set);
+
+GVAR(isCurator) = _isCurator;
+GVAR(currentAircraft) = _aircraft;
+
+createDialog QGVAR(DialogLoadout);
+private _display = DISPLAY(ID_DIALOG);
+_display displayAddEventHandler ["Unload", LINKFUNC(onButtonClose)];
+
+if (GVAR(rearmNewPylons) || {_isCurator}) then {
+    ctrlShow [ID_TEXT_BANNER, false];
+};
+
+private _config = configFile >> "CfgVehicles" >> typeOf _aircraft;
+private _pylonComponent = _config >> "Components" >> "TransportPylonsComponent";
+
+ctrlSetText [ID_PICTURE_AIRCRAFT, getText (_pylonComponent >> "uiPicture")];
+
+private _hasFRIES = getNumber (_config >> QEGVAR(fastroping,enabled));
+if (["ace_fastroping"] call EFUNC(common,isModLoaded) && {_hasFRIES > 1}) then {
+    private _checkbox = _display displayCtrl ID_CHECKBOX_FRIES;
+    private _fries = _aircraft getVariable [QEGVAR(fastroping,FRIES), objNull];
+    _checkbox cbSetChecked (!isNull _fries);
+    _checkbox setVariable [QGVAR(originalState), !isNull _fries];
+} else {
+    ctrlShow [ID_CHECKBOX_FRIES, false];
+    ctrlShow [ID_TEXT_FRIES, false];
+};
+
+GVAR(comboBoxes) = [];
+{
+    private _combo = _display ctrlCreate ["ctrlCombo", -1];
+    private _picturePos = ctrlPosition (_display displayCtrl ID_PICTURE_AIRCRAFT);
+    private _uiPos = getArray (_x >> "UIposition");
+    _combo ctrlSetPosition [
+        (_picturePos select 0) + (_uiPos select 0),
+        (_picturePos select 1) + (_uiPos select 1),
+        0.1 * safezoneW,
+        0.028 * safezoneH
+    ];
+    _combo ctrlCommit 0;
+
+    _combo lbAdd localize LSTRING(Empty);
+    _combo lbSetData [0, ""];
+
+    private _mag = (getPylonMagazines _aircraft) select _forEachIndex;
+    private _mags = _aircraft getCompatiblePylonMagazines (_forEachIndex + 1);
+    private _index = 0;
+    {
+        _combo lbAdd getText (configFile >> "CfgMagazines" >> _x >> "displayName");
+        _combo lbSetData [_forEachIndex + 1, _x];
+
+        if (_x == _mag) then {
+            _index = _forEachIndex + 1;
+        };
+    } forEach _mags;
+    _combo lbSetCurSel _index;
+    _combo ctrlAddEventHandler ["LBSelChanged", LINKFUNC(onComboSelChange)];
+    // TODO: Allow pylon priority selection?
+
+    private _mirroredIndex = getNumber (_x >> "mirroredMissilePos");
+
+    private _button = controlNull;
+    if (count allTurrets [_aircraft, false] > 0) then {
+        _button = _display ctrlCreate ["ctrlShortcutButton", -1];
```

Comment:
use `ctrlButtonPictureKeepAspect` and the `ctrlSetText` changes to prevent issues with ui scaling